src/Items/ItemEyeOfEnder.h

```diff
@@ -0,0 +1,59 @@
+
+#pragma once
+
+#include "ItemHandler.h"
+#include "ItemThrowable.h"
+
+
+
+
+
+class cItemEyeOfEnderHandler :
+	public cItemThrowableHandler
+{
+	typedef cItemThrowableHandler super;
+public:
+	cItemEyeOfEnderHandler(void) :
+		super(E_ITEM_EYE_OF_ENDER, cProjectileEntity::pkSnowball, 30)
+	{
+	}
+
+	virtual bool OnItemUse(
+		cWorld * a_World, cPlayer * a_Player, cBlockPluginInterface & a_PluginInterface, const cItem & a_Item,
+		int a_BlockX, int a_BlockY, int a_BlockZ, eBlockFace a_BlockFace
+	) override
+	{
+		BLOCKTYPE FacingBlock;
+		NIBBLETYPE FacingMeta;
+		a_World->GetBlockTypeMeta(a_BlockX, a_BlockY, a_BlockZ, FacingBlock, FacingMeta);
+		switch (FacingBlock)
+		{
+			case E_BLOCK_END_PORTAL_FRAME:
+			{
+				// Fill the portal frame. 0x4 is the bit for holding the eye of ender.
+				if ((FacingMeta & 4) != 4)
+				{
+					a_World->SetBlock(a_BlockX, a_BlockY, a_BlockZ, E_BLOCK_END_PORTAL_FRAME, FacingMeta | 4);
+					if (!a_Player->IsGameModeCreative())
```

Comment:
Is spectator mode relevant here?

Reply:
In spectator mode, you can't use items.

The item should be used up in Survival and Adventure modes.